pandas/core/indexes/datetimes.py

```diff
@@ -608,12 +608,14 @@ def _simple_new(cls, values, name=None, freq=None, tz=None,
                            dtype=dtype, **kwargs)
             values = np.array(values, copy=False)
 
-        if is_object_dtype(values):
-            return cls(values, name=name, freq=freq, tz=tz,
-                       dtype=dtype, **kwargs).values
-        elif not is_datetime64_dtype(values):
+        if not is_datetime64_dtype(values):
             values = _ensure_int64(values).view(_NS_DTYPE)
 
+        values = getattr(values, 'values', values)
```

Comment:
Why is this one still needed? I thought the idea was now to ensure ndarrays are passed to `_simple_new` and not DatetimeIndexes?

Reply:
Well-spotted! I inserted this when I was trying to investigate why some tests were failing and I meant to move it before pushing the commit but forgot. I think we can move this line to just before the call to ``_simple_new`` in this file:
https://github.com/pandas-dev/pandas/blob/1dd05ccf56b163dc51786e4f5ce76b279cc0dec3/pandas/core/arrays/datetimelike.py#L43-L53

Does this make sense? I did the same thing here:
https://github.com/pandas-dev/pandas/blob/1dd05ccf56b163dc51786e4f5ce76b279cc0dec3/pandas/core/indexes/base.py#L501-L513